Libraries/LibCrypto/Hash/HashManager.h

```diff
@@ -90,6 +90,11 @@ class Manager final : public HashFunction<0, 0, MultiHashDigestVariant> {
 public:
     using HashFunction::update;
 
+    static NonnullOwnPtr<Manager> create(HashKind kind)
```

Comment:
why is it necessary for the Manager class to be heap-allocated? it should be fine as a value type.

Reply:
Because it is sometimes used for the HMAC constructor which does:
```c++
    HMAC(KeyBufferType key, Args... args)
        : m_inner_hasher(move(HashT::create(args...)))
        , m_outer_hasher(move(HashT::create(args...)))
```

It doesn't need to be heap allocated, but it is required to be compatible with HMAC.